src/plugins/mer/mersdkdetailswidget.cpp

```diff
@@ -208,5 +234,142 @@ void MerSdkDetailsWidget::onPathChooserEditingFinished()
+        m_wwwProxyServers.clear();
+        m_ui->wwwProxyServersLine->clear();
+    }
+    m_ui->wwwProxyServersLine->setHidden(true);
+
+    m_ui->wwwProxyExcludesLabel->setHidden(true);
+    if (!m_ui->wwwProxyExcludesLine->text().isEmpty()) {
+        m_wwwProxyExcludes.clear();
+        m_ui->wwwProxyExcludesLine->clear();
+    }
+    m_ui->wwwProxyExcludesLine->setHidden(true);
+
+    emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServers, m_wwwProxyExcludes);
+}
+
+void MerSdkDetailsWidget::onWwwProxyAutomaticClicked()
+{
+    if (!m_ui->wwwProxyAutomaticButton->isChecked()
+            || m_wwwProxy == MER_SDK_PROXY_AUTOMATIC) {
+        return;
+    }
+
+    m_wwwProxy = MER_SDK_PROXY_AUTOMATIC;
+
+    m_ui->wwwProxyServersLabel->setText(tr("URL:"));
+    m_ui->wwwProxyServersLabel->setHidden(false);
+    if (!m_ui->wwwProxyServersLine->text().isEmpty()) {
+        m_wwwProxyServers.clear();
+        m_ui->wwwProxyServersLine->clear();
+    }
+    m_ui->wwwProxyServersLine->setHidden(false);
+    m_ui->wwwProxyServersLine->setToolTip(tr("Leave empty for asking the network, otherwise insert a proxy autoconfiguration URL."));
+
+    m_ui->wwwProxyExcludesLabel->setHidden(true);
+    if (!m_ui->wwwProxyExcludesLine->text().isEmpty()) {
+        m_wwwProxyExcludes.clear();
+        m_ui->wwwProxyExcludesLine->clear();
+    }
+    m_ui->wwwProxyExcludesLine->setHidden(true);
+
+    emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServers, m_wwwProxyExcludes);
+}
+
+void MerSdkDetailsWidget::onWwwProxyManualClicked()
+{
+    if (!m_ui->wwwProxyManualButton->isChecked()
+            || m_wwwProxy == MER_SDK_PROXY_MANUAL) {
+        return;
+    }
+
+    m_wwwProxy = MER_SDK_PROXY_MANUAL;
+
+    m_ui->wwwProxyServersLabel->setText(tr("Servers:"));
+    m_ui->wwwProxyServersLabel->setHidden(false);
+    if (!m_ui->wwwProxyServersLine->text().isEmpty()) {
+        m_wwwProxyServers.clear();
+        m_ui->wwwProxyServersLine->clear();
+    }
+    m_ui->wwwProxyServersLine->setHidden(false);
+    m_ui->wwwProxyServersLine->setToolTip(tr("Insert a list of proxies. E.g. \"proxy1.example.com 1.2.3.4:8080\""));
+
+    m_ui->wwwProxyExcludesLabel->setHidden(false);
+    if (!m_ui->wwwProxyExcludesLine->text().isEmpty()) {
+        m_wwwProxyExcludes.clear();
+        m_ui->wwwProxyExcludesLine->clear();
+    }
+    m_ui->wwwProxyExcludesLine->setHidden(false);
+    m_ui->wwwProxyExcludesLine->setToolTip(tr("Insert a list of addresses excluded from the proxy. E.g. \"internal.example.com 10.0.0.0/8\""));
+
+    emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServers, m_wwwProxyExcludes);
+}
+
+void MerSdkDetailsWidget::onWwwProxyServersEdited(const QString &s)
+{
+    QString servers = s;
+    QRegExp reg;
+
+    if (servers == m_wwwProxyServers) {
+        return;
+    }
+
+    if (m_wwwProxy == MER_SDK_PROXY_AUTOMATIC) {
+        reg.setPattern("^(\\S*)(.*)$");
+        servers.replace(reg, "\\1");
+        reg.setPattern("[,;]");
+        servers.replace(reg, "");
```

Comment:
I do understand the regexps, but I do not get the purpose :) Could you please add a comment?